docs/fsharp/whats-new/fsharp-9.md

```diff
@@ -0,0 +1,432 @@
+
+This is a more natural syntax compared to the previously available `builder { () }`.
+
+### Hash directives are allowed to take non-string arguments
+
+Hash directives for the compiler previously only allowed string arguments passed in quotes. Now, they can take any type of argument.
+
+Previously, you had:
+```fsharp
+#nowarn "0070"
+#time "on"```
+
+Now, you can write:
+```fsharp
+#nowarn 0070
+#time on```
+
+This also ties into the next two changes.
+
+### Extended #help directive in fsi to show documentation in the REPL
+
+The `#help` directive in F# Interactive now shows documentation for a given object or function, which you can now pass without quotes.
+
+<pre><code>&gt; #help List.map;;
+
+Description:
+Builds a new collection whose elements are the results of applying the given function
+to each of the elements of the collection.
+
+Parameters:
+- mapping: The function to transform elements from the input list.
+- list: The input list.
+Returns:
+The list of transformed elements.
+
+Examples:
+let inputs = [ "a"; "bbb"; "cc" ]
+
+inputs |> List.map (fun x -> x.Length)
+// Evaluates to [ 1; 3; 2 ]
+
+Full name: Microsoft.FSharp.Collections.ListModule.map
+Assembly: FSharp.Core.dll
+```
+
+### Allow #nowarn to support the FS prefix on error codes to disable warnings
+
+Previously, when you wanted to disable a warning and wrote `#nowarn "FS0057"`, you would get an `Invalid warning number 'FS0057'`. Even though the warning number is correct, it just wasn't supposed to have the `FS` prefix.
+
+Now, you won't have to spend time figuring that out because the warning numbers are accepted even with the prefix.
+
+All of these will now work:
+```fsharp
+#nowarn 57
+#nowarn 0057
+#nowarn FS0057
+
+#nowarn "57"
+#nowarn "0057"
+#nowarn "FS0057"
+```
+
+It's probably a good idea to stick to the same style throughout your project.
+
+### Warning about TailCall attribute on non-rec functions or let-bound values
+
+F# will now emit a warning when you put the `[<TailCall>]` attribute somewhere it doesn't belong. While it has no effect on what the code does, it could confuse someone reading it.
+
+For example, these will now emit a warning:
+```fsharp
+[&lt;TailCall&gt;]
+let someNonRecFun x = x + x
+
+[&lt;TailCall&gt;]
+let someX = 23
+
+[&lt;TailCall&gt;]
+let rec someRecLetBoundValue = nameof(someRecLetBoundValue)
+```
+
+### Enforce attribute targets
+
+The compiler now correctly enforces the `AttributeTargets` on let values, functions, union case declarations, implicit constructors, structs, and classes. This can prevent some hard-to-notice bugs, such as forgetting to add the unit argument to an Xunit test.
+
+Previously, you could have written:
+
+```fsharp
+[&lt;Fact&gt;]
+let ``this test always fails`` =
+  Assert.True(false)
+```
+
+Run the tests with `dotnet test` and they would pass. Since the test function is not actually a function, it is ignored by the test runner.
```

Comment:
```suggestion
When you ran the tests with `dotnet test`, they would pass. Since the test function is not actually a function, it was ignored by the test runner.
```